cni/pkg/ambient/server.go

```diff
@@ -187,6 +189,12 @@ func (s *Server) ReconcileZtunnel() error {
 	s.UpdateConfig()
 	if activePod == nil {
 		log.Infof("active ztunnel updated, no ztunnel running on the node")
+		if s.redirectMode == EbpfMode {
+			if err := s.delZtunnelEbpfOnNode(); err != nil {
```

Comment:
why not just move this into cleanupNode? its pretty hard to follow with all these `if EBPF` without them consolidated in just a few places. 

In the future someone will probably call cleanupNode and have broken code?